web/controllers/auth.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Dwylbot.Auth do
+  import Plug.Conn
+  import Phoenix.Controller
+  alias Dwylbot.Router.Helpers
+
+  def init(_opts) do
+  end
+
+  def call(conn, _opts) do
+    # this OR operator is used for our tests
+    # the test assign values directly to the conn instead of creating some fake sessions/cookies
+    # see Chapter 8 page 137 section Integration Tests of the Programmin Phoenix book
+    user = get_session(conn, :current_user) || conn.assigns[:current_user]
```

Comment:
At the moment I save all the information I need into the session under :current_user and not :user_id. Normally the user_id session is used to load the user data from Postgres but I don't save the user in Postgres yet.
In my tests I assign to the conn strucuture the user that's why I'm using the operator OR to determine how to retreive the user, from the session (real app) or if the session doesn't exist from the conn.assigns (tests)
```user = get_session(conn, :current_user) || conn.assigns[:current_user]```